keysign/util.py

```diff
@@ -29,13 +30,14 @@
 
 
 def mac_generate(key, data):
-    mac = hmac.new(key, data).hexdigest().upper()
+    mac = hmac.new(key, data, hashlib.sha256).hexdigest().upper()
     log.info("MAC of %r is %r", data[:20], mac[:20])
-    return mac
+    return mac[:20]
+
 
 def mac_verify(key, data, mac):
     computed_mac = mac_generate(key, data)
-    result = hmac.compare_digest(mac.upper(), computed_mac.upper())
```

Comment:
You are right, indeed it is a bit weird.
But if we truncate the mac_generate, then when we scan a qr we only have 20 chars of the hash and we need to set a limit to the compare or it will fails.
As an alternative I can do something like:
```
computed_mac = mac_generate(key, data)
truncated_mac = computed_mac[:20]
result = hmac.compare_digest(mac.upper(), truncated_mac.upper())
```

If we want a variable truncation we can add it as a parameter, or optional parameter with default of 20.
Anyway as I said earlier, I'm also open to completely remove the truncation.